src/scipp/plt/model.py

```diff
@@ -0,0 +1,89 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2022 Scipp contributors (https://github.com/scipp)
+
+import uuid
+
+
+def _make_graphviz_digraph(*args, **kwargs):
+    try:
+        from graphviz import Digraph
+    except ImportError:
+        raise RuntimeError(
+            "Failed to import `graphviz`. "
+            "Use `pip install graphviz` (requires installed `graphviz` executable) or "
+            "`conda install -c conda-forge python-graphviz`.")
+    return Digraph(*args, **kwargs)
+
+
+def _add_graph_edges(dot, node, inventory, hide_views):
+    name = str(node.func)
+    inventory.append(name)
+    for child in node.children:
+        key = str(child.func)
+        if key not in inventory:
+            dot.edge(name, key)
+            _add_graph_edges(dot, child, inventory, hide_views)
+    for parent in node.parents + list(node.kwparents.values()):
+        key = str(parent.func)
+        if key not in inventory:
+            dot.edge(key, name)
+            _add_graph_edges(dot, parent, inventory, hide_views)
+    if not hide_views:
+        for view in node.views:
+            key = str(view)
+            dot.node(key, shape='ellipse', style='filled', color='lightgrey')
+            dot.edge(name, key)
+
+
+def show_graph(node, size=None, hide_views=False):
+    dot = _make_graphviz_digraph(strict=True)
+    dot.attr('node', shape='box', height='0.1')
+    dot.attr(size=size)
+    inventory = []
+    _add_graph_edges(dot, node, inventory, hide_views)
+    return dot
+
+
+class Node:
+
+    def __init__(self, func, *parents, **kwparents):
+        self.id = str(uuid.uuid1())
+        self.children = []
+        self.views = []
+        self.func = func if callable(func) else lambda: func
```

Comment:
Seems a bit hard to understand (from the user interface) how this works, calling `Node(da)`. Can you just require always passing a callable here and instead make a helper function to do this? Something like:

```python
    n = input_node(da)
    fig = Figure(n)
```
With
```python
def input_node(x):
  return x
```
?